homeassistant/components/sabnzbd/__init__.py

```diff
@@ -1,234 +1,58 @@
-DEFAULT_SPEED_LIMIT = "100"
-DEFAULT_SSL = False
-
-UPDATE_INTERVAL = timedelta(seconds=30)
-
-SERVICE_PAUSE = "pause"
-SERVICE_RESUME = "resume"
-SERVICE_SET_SPEED = "set_speed"
-
-SIGNAL_SABNZBD_UPDATED = "sabnzbd_updated"
 
-
-@dataclass
-class SabnzbdRequiredKeysMixin:
-    """Mixin for required keys."""
-
-    field_name: str
-
-
-@dataclass
-class SabnzbdSensorEntityDescription(SensorEntityDescription, SabnzbdRequiredKeysMixin):
-    """Describes Sabnzbd sensor entity."""
-
-
-SENSOR_TYPES: tuple[SabnzbdSensorEntityDescription, ...] = (
-    SabnzbdSensorEntityDescription(
-        key="current_status",
-        name="Status",
-        field_name="status",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="speed",
-        name="Speed",
-        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
-        field_name="kbpersec",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="queue_size",
-        name="Queue",
-        native_unit_of_measurement=DATA_MEGABYTES,
-        field_name="mb",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="queue_remaining",
-        name="Left",
-        native_unit_of_measurement=DATA_MEGABYTES,
-        field_name="mbleft",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="disk_size",
-        name="Disk",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="diskspacetotal1",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="disk_free",
-        name="Disk Free",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="diskspace1",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="queue_count",
-        name="Queue Count",
-        field_name="noofslots_total",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="day_size",
-        name="Daily Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="day_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="week_size",
-        name="Weekly Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="week_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="month_size",
-        name="Monthly Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="month_size",
-    ),
-    SabnzbdSensorEntityDescription(
-        key="total_size",
-        name="Total",
-        native_unit_of_measurement=DATA_GIGABYTES,
-        field_name="total_size",
-    ),
+from .const import (
+    ATTR_SPEED,
+    DEFAULT_SPEED_LIMIT,
+    DOMAIN,
+    KEY_API,
+    KEY_NAME,
+    SERVICE_PAUSE,
+    SERVICE_RESUME,
+    SERVICE_SET_SPEED,
+    SIGNAL_SABNZBD_UPDATED,
+    UPDATE_INTERVAL,
 )
+from .sab import get_client
 
-SENSOR_KEYS: list[str] = [desc.key for desc in SENSOR_TYPES]
+PLATFORMS = ["sensor"]
+_LOGGER = logging.getLogger(__name__)
 
 SPEED_LIMIT_SCHEMA = vol.Schema(
     {vol.Optional(ATTR_SPEED, default=DEFAULT_SPEED_LIMIT): cv.string}
 )
 
-CONFIG_SCHEMA = vol.Schema(
```

Comment:
Keep this and import it into a config entry.

Reply:
I'm not sure how to do this

Reply:
Search how `SOURCE_IMPORT` is used in `__init__.py` together with `async_step_import` in `config_flow.py` in other integrations

Reply:
I moved the vol.Schema to `config_flow.py` under the name `USER_SCHEMA`

Reply:
You need ot keep `CONFIG_SCHEMA`. It's a special name that Home Assistant looks up and makes sure it's correct before passing it to `async_setup`.

Reply:
Done